projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts

```diff
@@ -10,37 +10,58 @@ import { EMPTY, Observable, of } from 'rxjs';
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
     <div class="call-headers">
-      <ht-label label="Headers" class="title"></ht-label>
+      <ht-label [label]="this.label" class="title"></ht-label>
       <div class="container" data-sensitive-pii>
-        <ng-container *htLoadAsync="this.headerRecords$ as headerRecords">
-          <ht-list-view [records]="headerRecords"></ht-list-view>
+        <ng-container *htLoadAsync="this.records$ as records">
+          <ht-list-view [records]="records"></ht-list-view>
         </ng-container>
       </div>
     </div>
   `
 })
 export class SpanDetailCallHeadersComponent implements OnChanges {
   @Input()
-  public headers?: Dictionary<unknown>;
+  public data?: Dictionary<unknown>;
 
-  public headerRecords$?: Observable<ListViewRecord[]>;
+  @Input()
+  public mode: SpanDetailCallHeaderType = SpanDetailCallHeaderType.Header;
+
+  public records$?: Observable<ListViewRecord[]>;
+
+  public label?: string;
 
   public ngOnChanges(): void {
-    this.buildHeaderRecords();
+    this.label = this.getLabel();
+    this.buildRecords();
+  }
+
+  private getLabel(): string {
+    switch (this.mode) {
```

Comment:
nit: if this enum is only used for the title, it'd be better to just take the title as an input (and make this named something more generic).